src/relion/zocalo/wrapper.py

```diff
@@ -220,6 +232,20 @@ def start_relion(self):
         logger.info("Done.")
         return success
 
+    def check_whether_ended(self, proj):
+        if len(proj._job_nodes) == 0 or None in [j.attributes["status"] for j in proj]:
+            return False
+        check_time = time.time()
+        if all(
+            [
+                check_time - datetime.datetime.timestamp(j.attributes["end_time_stamp"])
+                > 10 * 60
+                for j in proj
+            ]
+        ):
+            return True
+        return False
```

Comment:
```suggestion
        return all(
            check_time - datetime.datetime.timestamp(j.attributes["end_time_stamp"])
            > 10 * 60
            for j in proj
        )
```
You can use generators directly in an `all()` context, which also means it will stop evaluating on the first false-y value it encounters.